src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+function deleteRedundantNumber(numberWithRedundantNumber: number): number {
+  return Math.floor(numberWithRedundantNumber * 100) / 100;
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const maxSide = Math.max(this.a, this.b, this.c);
+
+    if (
+      [this.a, this.b, this.c].some((item) => item <= 0) ||
+      maxSide >= this.a + this.b + this.c - maxSide
```

Comment:
You do not need to manually assign constructor parameters to class properties when using access modifiers. TypeScript does this automatically. You can remove the assignments for 'color', 'a', 'b', and 'c' parameters.

```suggestion
    const maxSide = Math.max(a, b, c);

    if (
      [a, b, c].some((item) => item <= 0) ||
      maxSide >= a + b + c - maxSide
```